src/lib/y2storage/proposal/lvm_creator.rb

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/planned"
+require "y2storage/disk_size"
+
+module Y2Storage
+  module Proposal
+    # Class to create LVM volume groups and logical volumes following a
+    # Planned::LvmVg object
+    class LvmCreator
+      include Yast::Logger
+
+      # Default name for volume groups
+      DEFAULT_VG_NAME = "system".freeze
+      # Default name for logical volumes
+      DEFAULT_LV_NAME = "lv".freeze
+
+      private_constant :DEFAULT_VG_NAME, :DEFAULT_LV_NAME
+
+      # @return [Devicegraph] initial devicegraph
+      attr_reader :original_devicegraph
+
+      # Constructor
+      #
+      # @param original_devicegraph [Devicegraph] Initial devicegraph
+      def initialize(original_devicegraph)
+        @original_devicegraph = original_devicegraph
+      end
+
+      # Returns a copy of the original devicegraph in which the volume
+      # group (and its logical volumes, if needed) have been created.
+      #
+      # @param planned_vg [Planned::LvmVg]   Volume group to create
+      # @param pv_partitions [Array<String>] names of the newly created
+      #     partitions that should be added as PVs to the volume group
+      # @return [Devicegraph] New devicegraph containing the planned volume group
+      def create_volumes(planned_vg, pv_partitions = [])
```

Comment:
We should use `planned_vg.pvs` instead of an additional `pv_partitions`  argument.